apps/webapp/web/static/js/docs/index.js

```diff
@@ -24,24 +25,42 @@ class Index extends React.Component {
         return response.json();
       })
       .then(list => {
-        this.setState({ docs: list });
+        this.setState({ docs: list.data });
+      });
+
+    fetch(window.location.origin + '/docs/use_case_examples.md')
+      .then(response => {
+        if (response.status >= 400) {
+          throw new Error("Bad response from server");
+        }
+        return response.json();
+      })
+      .then(doc => {
+        this.setState({ file: doc.data });
       });
   }
 
   changeDocument (e) {
     e.preventDefault();
-    this.setState({
-      path: e.target.getAttribute('data-url')
-    })
+    fetch(window.location.origin + '/docs/' + e.target.getAttribute('data-filename'))
+      .then(response => {
+        if (response.status >= 400) {
+          throw new Error("Bad response from server");
+        }
+        return response.json();
+      })
+      .then(doc => {
+        this.setState({ file: doc.data });
```

Comment:
Here all the `fetch()`s have the following thing in common:

1. They all have the same "error" handling. _Maybe you could be more specific_.
2. They all have the same `location.origin`. _This is the BASE_URL and should be changed by a var set from the server_
3. They all only differ in the path and the _success_ handler.

Having said that, I'd implement something like:

```javascript
fetchDocsData(path) {
  //DOCS_BASE_URL should end with / so path don't need to start with /
  return fetch(`DOCS_BASE_URL${path}`) 
    .then(/*handle 400*/)
    .catch(/*handle other errors such as 500*/)
}
```
Then you'd use this function like:

```javascript
fetchDocsData('index').then(response => this.setState({docs: response.data}))
// or
fetchDocsData(docPath).then(response => this.setState({file: response.data}))
```

For more info take a look at docs about Javascript Promise and Fetch APIs